### ISSUE TITLE
Metainfo type size not correct? (DeepState fuzzing)

### ISSUE THREAD

**Issue by agroce** [Blosc/c-blosc2#92]
I just started fuzzing C-Blosc2 with DeepState.  The harness is here: https://github.com/agroce/deepstate-c-blosc2

A few odd things are coming up immediately.  No round-trip failures yet, but oddities, maybe API misunderstandings.

```
    size_t typesize;
    int flags;
    blosc_cbuffer_metainfo(intermediate, &typesize, &flags);
    LOG(TRACE) << "type size = " << typesize;
    //ASSERT_EQ(typesize, type_size) << "type size = " << type_size << " but meta claims " << typesize;
```

The returned `typesize` seems to always be 1 in runs, whatever it was when the call to compress was made.

**Comment by agroce** [comment]
Looks like this is due to (undocumented?) fact type_size shouldn't be bigger than 255

**Comment by FrancescAlted** [comment]
Right.  When `typesize` is > 255, then typesize is [silently dropped to 1](https://github.com/Blosc/c-blosc2/blob/master/blosc/blosc2.c#L1345).  The rational for this is that shuffling is a very costly operation for large typesizes, so dropping to 1 is a good way to be fast again (at the expense of some compression ratio drop; or not, depends on the dataset).  Agreed that this should be documented.
